packages/auth/amplify_auth_cognito_dart/lib/src/model/attribute/cognito_resend_user_attribute_confirmation_code_options.dart

```diff
@@ -5,39 +5,87 @@ import 'package:amplify_core/amplify_core.dart';
 
 part 'cognito_resend_user_attribute_confirmation_code_options.g.dart';
 
+const _deprecatedMessage = '''
+Use ResendUserAttributeConfirmationCodeOptions instead. If Cognito-specific options are needed, use `pluginOptions`:
+
+ResendUserAttributeConfirmationCodeOptions(
+  pluginOptions: CognitoResendUserAttributeConfirmationCodePluginOptions(
+    clientMetadata: {
+      'my-key': 'my-value',
+    },
+  ),
+)
+''';
+
 /// {@template amplify_auth_cognito.cognito_resend_user_attribute_confirmation_code_options}
-/// Cognito extension of [ResendUserAttributeConfirmationCodeOptions] to add the
-/// platform specific fields.
+/// Cognito options for `Amplify.Auth.resendUserAttributeConfirmationCode`.
 /// {@endtemplate}
 @zAmplifySerializable
+@Deprecated(_deprecatedMessage)
 class CognitoResendUserAttributeConfirmationCodeOptions
-    extends ResendUserAttributeConfirmationCodeOptions
-    with
-        AWSEquatable<CognitoResendUserAttributeConfirmationCodeOptions>,
-        AWSDebuggable {
+    extends ResendUserAttributeConfirmationCodeOptions {
   /// {@macro amplify_auth_cognito.cognito_resend_user_attribute_confirmation_code_options}
+  @Deprecated(_deprecatedMessage)
   const CognitoResendUserAttributeConfirmationCodeOptions({
-    this.clientMetadata,
-  });
+    Map<String, String>? clientMetadata,
+  }) : clientMetadata = clientMetadata ?? const {};
 
   /// {@macro amplify_auth_cognito.cognito_resend_user_attribute_confirmation_code_options}
+  @Deprecated(_deprecatedMessage)
   factory CognitoResendUserAttributeConfirmationCodeOptions.fromJson(
     Map<String, Object?> json,
   ) =>
       _$CognitoResendUserAttributeConfirmationCodeOptionsFromJson(json);
 
+  /// {@macro amplify_auth_cognito.model.cognito_resend_user_attribute_confirmation_code_plugin_options.client_metadata}
+  final Map<String, String> clientMetadata;
+
+  @override
+  CognitoResendUserAttributeConfirmationCodePluginOptions get pluginOptions =>
+      CognitoResendUserAttributeConfirmationCodePluginOptions(
+        clientMetadata: clientMetadata,
+      );
+
+  @override
+  String get runtimeTypeName =>
+      'CognitoResendUserAttributeConfirmationCodeOptions';
+
+  @override
+  Map<String, Object?> toJson() =>
+      _$CognitoResendUserAttributeConfirmationCodeOptionsToJson(this);
+}
+
+/// {@template amplify_auth_cognito.model.cognito_resend_user_attribute_confirmation_code_plugin_options}
+/// Cognito options for `Amplify.Auth.resendUserAttributeConfirmationCode`.
+/// {@endtemplate}
+@zAmplifySerializable
+class CognitoResendUserAttributeConfirmationCodePluginOptions
```

Comment:
how about to move PluginOptions classes to their own file so that they follow the class -> file naming convention. I mean instead of having `CognitoResendUserAttributeConfirmationCodePluginOptions` defined in `cognito_resend_user_attribute_confirmation_code_options.dart` it will be defined in `cognito_resend_user_attribute_confirmation_code_plugin_options.dart`

Reply:
Hm do you think we should do the same for the category plugin options as well?

Reply:
I don't think so bc they are abstract classes and used only by category level options and so it is okay to have them inside the options file.